### PR TITLE
Add a lint rule that detects truncated number literals.

### PR DESCRIPTION
**Pull request by hzeller**
Detects number literals that have a shorter bit-width than the
digits provided, e.g. 16'h1FFFF.

Decimal numbers are only dealt with correctly up to 128 bits,
beyond that, only some lower-bound estimates are applied so
it might miss complain opportunity there. Given that super-long
constants are very likely not given as decimal, this
tradeoff is accepted.

Signed-off-by: Henner Zeller <hzeller@google.com>

### CODE REVIEW

File: verilog/analysis/default_rules.h [hunk 1]
```diff
@@ -65,6 +65,7 @@ constexpr const char* kDefaultRuleSet[] = {
     "positive-meaning-parameter-name",
     "constraint-name-style",
     "suggest-parentheses",
+    "truncated-numeric-literal",
```
Comment (fangism) [inline verilog/analysis/default_rules.h:68]: I used to recommend flipping the default switch separately to make it easily revertible.

**Review by fangism — APPROVED** [review]
Awesome test coverage of corner cases!

File: verilog/analysis/default_rules.h (hunk 1 above)
Comment (hzeller) [reply verilog/analysis/default_rules.h:68]: sounds good, will comment here and make separate PR with just that flip.
